php/stats.php

```diff
@@ -1,10 +1,15 @@
 <?php
 
-include_once 'locale.php';
+session_start();
+
+include_once 'config.php';
 include_once 'db_pdo.php';
 include_once 'helper.php';
 include_once 'filter.php';
 
+// Mark strings as localizable but don't localize.
+function _($str) { return $str; }
```

Comment:
This actually causes breakages on my dev install (and therefore presumably would also in production).

```
"PHP message: PHP Fatal error:  Cannot redeclare _() in /var/www/openflights/php/stats.php on line 11
```

But I haven't explicitly installed the PHP gettext extension, I believe. Just gettext for the CLI commands...

```
# dpkg -S /usr/lib/php/20220829/gettext.so 
php8.2-common: /usr/lib/php/20220829/gettext.so
# dpkg -S /usr/lib/php/20190902/gettext.so 
php7.4-common: /usr/lib/php/20190902/gettext.so
```

And that's from the sury packages..

```
# dpkg -l | grep php | grep common
ii  php-common                            2:93+ubuntu22.04.1+deb.sury.org+2            all          Common files for PHP packages
ii  php7.4-common                         1:7.4.33-6+ubuntu22.04.1+deb.sury.org+1      amd64        documentation, examples and common module for PHP
ii  php8.2-common                         8.2.8-1+ubuntu22.04.1+deb.sury.org+1         amd64        documentation, examples and common module for PHP
```

I did set `$OF_USE_LOCALES = true;` to be able to test things.

If I set `$OF_USE_LOCALES = false;`, I get

```
PHP message: PHP Fatal error:  Cannot redeclare _() in /var/www/openflights/php/locale.php on line 28
```
Looking at my bash history, I don't seem to have explicitly installed it either.

But looking again at the INSTALL and #1206 neither explicitly install any of the `php-common` packages...

https://packages.ubuntu.com/jammy/amd64/php8.1 depends on https://packages.ubuntu.com/jammy/amd64/php8.1-common which depends on https://packages.ubuntu.com/jammy/amd64/php-common

```
# apt-cache --installed rdepends php7.4-common
php7.4-common
Reverse Depends:
  php7.4-zip
  php7.4-xml
  php7.4-readline
  php7.4-opcache
  php7.4-mysql
  php7.4-mbstring
  php7.4-json
  php7.4-gd
  php7.4-fpm
  php7.4-curl
  php7.4-cli
  php7.4
  php-common
# apt-cache --installed rdepends php8.2-common
php8.2-common
Reverse Depends:
  php8.2-zip
  php8.2-xml
  php8.2-xdebug
  php8.2-readline
  php8.2-opcache
  php8.2-mysql
  php8.2-mbstring
  php8.2-gd
  php8.2-fpm
  php8.2-curl
  php8.2-cli
# apt-cache --installed rdepends php-common
php-common
Reverse Depends:
  php8.2-zip
  php8.2-xml
  php8.2-xdebug
  php8.2-readline
  php8.2-opcache
  php8.2-mysql
  php8.2-mbstring
  php8.2-gd
  php8.2-curl
  php8.2-common
  php7.4-zip
  php7.4-xml
  php7.4-readline
  php7.4-opcache
  php7.4-mysql
  php7.4-mbstring
  php7.4-json
  php7.4-gd
  php7.4-curl
  php7.4-common
  php-zip
  php-xdebug
  php-mbstring
  php-json
  php-zip
  php-xdebug
  php-mbstring
  php-json
```

So you should have installed it on your Docker install too...

I'm guessing either something has changed with packaging (since whatever version the site was previously running on), as it's there in the default package on [22.04](https://packages.ubuntu.com/jammy/amd64/php8.1-common/filelist) (might need to refresh a few times, `packages.ubuntu.com` is giving intermittant HTTP 500)...

Does it work for you?

What do you have `$OF_USE_LOCALES` set to?

As far as I can tell... any calls to `php/stats.php` will be broken whichever way you've got it set...

And almost any other request (just loading the site) will be broken if you have `$OF_USE_LOCALES = false;` too...

Reply:
Oof, yeah, it doesn't work. I admit I assumed it was fine based on `locale.php` already doing it. Ugh. And turning off locales breaks my local version as well.

Reply:
I dunno if `$OF_USE_LOCALES` should be deprecated/removed, based on the existence of the apparent existence in default PHP...

Not stuff for this commit, but:

https://github.com/jpatokal/openflights/blob/5e0fae7fa7d44e4414d3ea28b585da1bc5c2712b/php/config.php.sample#L21 wants updating...

composer runs would actually fail if the "extension" wasn't loaded (unless you tried to use `--ignore-platform-reqs`) - https://github.com/jpatokal/openflights/blob/5e0fae7fa7d44e4414d3ea28b585da1bc5c2712b/composer.lock#L4346

https://github.com/jpatokal/openflights/blob/0f1adfc/php/locale.php#L28-L30 should either be removed (because of the composer requirement), or guarded with a `!function_exists('_(')` call for it's definition (I doubt it's possible for `ext-gettext` to be apparently existing (ignoring `--ignore-platform-reqs`), but the function to not)...

Some distro's do do some weird stuff regarding packaging of php extensions... Some are included, some are pulled out etc...

But either way, both shouldn't be able to be true.

`$OF_USE_LOCALES` maybe/probably still has some value, basically unconditionally forcing `en_US` for all requests...

Reply:
Heh, the `config.php.sample` comment is from c7b2cdf3967c8471eeddf284a45a038aba64278d (May 2013)...

Reply:
Can you just set `$locale = "en_US.utf8";` at the top of the file?

Reply:
> Can you just set `$locale = "en_US.utf8";` at the top of the file?
> 
> Which, in theory, should "force" "en_US" for all `_()` calls in the file...

How would that work? Wouldn't `locale.php` override it with whatever is in `$_SESSION`?